main.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
+	k8sReport "github.com/aquasecurity/trivy/pkg/k8s/report"
+	"github.com/aquasecurity/trivy/pkg/report"
+	"github.com/aquasecurity/trivy/pkg/types"
+)
+
+var (
+	tempJsonFileName = "csv-report-temp.json"
+	templateFileName = "csv.tpl"
+)
+
+func init() {
+	var CustomTemplateFuncMap = map[string]interface{}{
+		"escapeCsv": func(input string) string {
+			quoted := strconv.Quote(input)
+			return strings.ReplaceAll(quoted, "\\\"", "\"\"")
+		},
+		"escapeString": func(input string) dbTypes.SourceID {
+			return dbTypes.SourceID(input)
+		},
+		"nvdV3Score": func(input dbTypes.VendorCVSS) float64 {
+			return input["nvd"].V3Score
+		},
+		"rhV3Score": func(input dbTypes.VendorCVSS) float64 {
+			return input["redhat"].V3Score
+		},
+		"nvdV3Vector": func(input dbTypes.VendorCVSS) string {
+			return input["nvd"].V3Vector
+		},
+		"rhV3Vector": func(input dbTypes.VendorCVSS) string {
+			return input["redhat"].V3Vector
+		},
+	}
+	report.CustomTemplateFuncMap = CustomTemplateFuncMap
+}
+func main() {
+	trivyCommand := os.Args[1 : len(os.Args)-1]
+	scanType := trivyCommand[0]
+	outputFileName := os.Args[len(os.Args)-1]
+	cmdArgs := append(trivyCommand, "--format", "json", "--output", tempJsonFileName)
```

Comment:
we can store this temporary file in the temporary folder ('os.TempDir())'), then  we don't need to use `joinBaseDir` for it. right?